airframe/src/test/scala/wvlet/airframe/CloseableShutdownHookTest.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package wvlet.airframe
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import wvlet.airspec.AirSpec
+
+/**
+  *
+  */
+class CloseableShutdownHookTest extends AirSpec {
+  scalaJsSupport
+
+  class A extends AutoCloseable {
```

Comment:
@takezoe Note. with the support for inner classes in Surface, we can define a test class closer in a closer location to the test method.